Disable CPU optimizations not supported by all x86_64 systems. Also
don't add anything for i686.

--- a/cmake/modules/SIMDExt.cmake	2017-03-23 22:22:58.254071694 +0100
+++ b/cmake/modules/SIMDExt.cmake	2017-03-23 22:23:22.446848845 +0100
@@ -5,11 +5,6 @@
 # HAVE_ARM_NEON
 # HAVE_INTEL_SSE
 # HAVE_INTEL_SSE2
-# HAVE_INTEL_SSE3
-# HAVE_INTEL_SSSE3
-# HAVE_INTEL_PCLMUL
-# HAVE_INTEL_SSE4_1
-# HAVE_INTEL_SSE4_2
 #
 # SIMD_COMPILE_FLAGS
 #
@@ -56,7 +51,7 @@
   if(HAVE_ARM_NEON)
     set(SIMD_COMPILE_FLAGS "${SIMD_COMPILE_FLAGS} -mfpu=neon")
   endif()
-elseif(CMAKE_SYSTEM_PROCESSOR MATCHES "i386|i686|amd64|x86_64|AMD64")
+elseif(CMAKE_SYSTEM_PROCESSOR MATCHES "amd64|x86_64|AMD64")
   set(HAVE_INTEL 1)
   CHECK_C_COMPILER_FLAG(-msse HAVE_INTEL_SSE)
   if(HAVE_INTEL_SSE)
@@ -66,24 +61,4 @@
   if(HAVE_INTEL_SSE2)
     set(SIMD_COMPILE_FLAGS "${SIMD_COMPILE_FLAGS} -msse2")
   endif()
-  CHECK_C_COMPILER_FLAG(-msse3 HAVE_INTEL_SSE3)
-  if(HAVE_INTEL_SSE3)
-    set(SIMD_COMPILE_FLAGS "${SIMD_COMPILE_FLAGS} -msse3")
-  endif()
-  CHECK_C_COMPILER_FLAG(-mssse3 HAVE_INTEL_SSSE3)
-  if(HAVE_INTEL_SSSE3)
-    set(SIMD_COMPILE_FLAGS "${SIMD_COMPILE_FLAGS} -mssse3")
-  endif()
-  CHECK_C_COMPILER_FLAG(-mpclmul HAVE_INTEL_PCLMUL)
-  if(HAVE_INTEL_PCLMUL)
-    set(SIMD_COMPILE_FLAGS "${SIMD_COMPILE_FLAGS} -mpclmul")
-  endif()
-  CHECK_C_COMPILER_FLAG(-msse4.1 HAVE_INTEL_SSE4_1)
-  if(HAVE_INTEL_SSE4_1)
-    set(SIMD_COMPILE_FLAGS "${SIMD_COMPILE_FLAGS} -msse4.1")
-  endif()
-  CHECK_C_COMPILER_FLAG(-msse4.2 HAVE_INTEL_SSE4_2)
-  if(HAVE_INTEL_SSE4_2)
-    set(SIMD_COMPILE_FLAGS "${SIMD_COMPILE_FLAGS} -msse4.2")
-  endif()
 endif()
